bambi/utils.py

```diff
@@ -13,46 +24,46 @@ def listify(obj):
 
 
 def extract_family_prior(family, priors):
-    # Given a family and a dictionary of priors, return the prior that could be
-    # passed to update the prior of a parameter in that family.
-
-    # Only built-in gaussian and negativebinomial have nuisance parameters
-    # i.e, parameters not related to the link-transformed predicted outcome
-    if isinstance(family, str):
-        if family == "gaussian" and "sigma" in priors:
-            return {"sigma": priors["sigma"]}
-        elif family == "negativebinomial" and "alpha" in priors:
-            return {"alpha": priors["alpha"]}
-        elif family == "gamma" and "alpha" in priors:
-            return {"alpha": priors["alpha"]}
+    """Extract priors for a given family
+
+    If a key in the priors dictionary matches the name of a nuisance parameter of the response
+    distribution for the given family, this function extracts and returns the prior for that
+    nuisance parameter. The result of this function can be safely used to update the ``Prior`` of
+    the response term.
+
+    Parameters
+    ----------
+    family: str or ``Family``
+        The name of a built-in family or a ``Family`` instance.
+    priors: dict
+        A dictionary where keys represent parameter/term names and values represent
+        prior distributions.
+    """
+
+    if isinstance(family, str) and family in FAMILY_PARAMS:
+        name = FAMILY_PARAMS[family]
+        prior = priors.pop(name, None)
+        if prior:
+            return {name: prior}
     elif isinstance(family, Family):
         # Only work if there are nuisance parameters in the family, and if any of these nuisance
         # parameters is present in 'priors' dictionary.
         nuisance_params = [k for k in family.prior.args if k not in ["observed", family.parent]]
         if set(nuisance_params).intersection(set(priors)):
-            return {k: priors[k] for k in nuisance_params if k in priors}
+            return {k: priors.pop(k) for k in nuisance_params if k in priors}
     return None
```

Comment:
What happens if none is returned to downstream call?

Reply:
Nothing. When None is returned it means the user didn't pass any prior for any parameter in the response distribution and there's no need to update them.